src/Kiota.Builder/Writers/Php/CodeMethodWriter.cs

```diff
@@ -36,7 +40,7 @@ public override void  WriteCodeElement(CodeMethod codeElement, LanguageWriter wr
                         WriteConstructorBody(parentClass, codeElement, writer, inherits);
                         break;
                     case CodeMethodKind.Serializer:
-                        WriteSerializerBody(codeElement, parentClass, writer, inherits);
+                        WriteSerializerBody(parentClass, writer, inherits, extendsModelClass);
```

Comment:
Since the methods downstream just end up using `inherits && extendsModelClass` to make decisions, I don't think we need to add the extra parameter to the methods but just pass `extendsModelClass` as it already it already captures `inherits` in the condition.